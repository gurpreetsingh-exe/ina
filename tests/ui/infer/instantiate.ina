// COMPILE_FAIL

fn app[T, U](x: T, f: fn(T) -> U) -> U {
    f(x)
}

fn id[T](x: T) -> T { x }

fn main() {
    // OK
    app(3, id);
    // ERROR
    app![i32, i64](3, id);
}
