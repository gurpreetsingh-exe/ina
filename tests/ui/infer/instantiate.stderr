error: mismatch types
    ╭─[tests/ui/infer/instantiate.ina:13:23]
    │
 13 │     app![i32, i64](3, id);
    │                       ╶╮
    │                        ╰── expected `i64`, found `i32`
    ∙
────╯
